samples/client/petstore/swift5/urlsessionLibrary/Sources/PetstoreClient/APIs/StoreAPI.swift

```diff
@@ -102,7 +102,7 @@ open class StoreAPI {
 
         let localVariableHeaderParameters = APIHelper.rejectNilHeaders(localVariableNillableHeaders)
 
-        let localVariableRequestBuilder: RequestBuilder<[String: Int]>.Type = PetstoreClientAPI.requestBuilderFactory.getBuilder()
+        let localVariableRequestBuilder: RequestBuilder<[String: Int]>.Type = PetstoreClientAPI.requestBuilderFactory.getBuilderAuthenticated()
 
         return localVariableRequestBuilder.init(method: "GET", URLString: (localVariableUrlComponents?.string ?? localVariableURLString), parameters: localVariableParameters, headers: localVariableHeaderParameters)
```

Comment:
@Jonas1893 Hey, thanks for creating this PR.
This is an issue that some people have, so thanks for taking the time to improve the generator.

Could you please provide an example of an implementation of `getDecodableBuilderAuthenticated`?

This solution is nice, but a bit verbose, because we need to provide 4 different builder implementations.

Did you consider alternatives?
For example passing a bool requiresAuthentication to this method?
By doing this, we only need the current 2 builders, so it's not a breaking change, and the we can get the endpoint configuration without doing any request.

Reply:
Hi @4brunu. I updated the `BearerRequestBuilderFactory` implementations in the PetStore samples to show the difference. The *authenticated builder version stays the same as before, always injecting a token into the request. For the unauthenticated builder we just use the already provided default `AlamofireRequestBuilder`s that does not inject anything. 

I did consider the alternative of distinguishing via parameter, however still this would result in a breaking API change - as long as we don't use a default parameter which I don't think we should, as this would be "guessing" and it might lead to bugs later if the default behavior is forgotten:

```
internal protocol RequestBuilderFactory {
    func getNonDecodableBuilder<T>() -> RequestBuilder<T>.Type
    func getBuilder<T: Decodable>() -> RequestBuilder<T>.Type
}
```
-> 
```
internal protocol RequestBuilderFactory {
    func getNonDecodableBuilder<T>(isRequestAuthenticated: Bool) -> RequestBuilder<T>.Type
    func getBuilder<T: Decodable>(isRequestAuthenticated: Bool) -> RequestBuilder<T>.Type
}
```

Adoption on app side would result in a if else in each method, e.g.:
```
func getBuilder<T: Decodable>(isRequestAuthenticated: Bool) -> RequestBuilder<T>.Type {
    if isRequestAuthenticated {
        return BearerRequestBuilder<T>.self
    } else {
        return AlamofireRequestBuilder<T>.self
    }
}
```

At this point, I don't think there is any benefit to either approach and we can just chose whatever we feel looks nicer. If you prefer the parameter approach I can also adjust it that way, no problem

Reply:
Hi @Jonas1893, sorry if I miscommunicated.
I was not suggesting passing the bool to the `getBuilder` function.
I was suggesting passing it to the init of the the `RequestBuilder`.


```
return localVariableRequestBuilder.init(method: "GET", URLString: (localVariableUrlComponents?.string ?? localVariableURLString), parameters: localVariableParameters, headers: localVariableHeaderParameters, requiresAuthentication: true)
``` 

This way it's not a breaking change, and we change even check if a specific endpoint requires authentication or not by calling the `getPetByIdWithRequestBuilder()`, and we only need the current 2 builders.

Reply:
@4brunu that is an excellent point. I created a new PR #13321 to be able to compare the two approaches.